src/main/java/com/epam/upskillproject/connect/customds/CustomPooledDataSource.java

```
@@ -147,23 +151,27 @@ public void setInactivityTimeLimitMillis(int inactivityTimeLimitMillis) throws I
     }
 
     @Override
-    public synchronized Connection getConnection() throws SQLException {
-        Connection connection = null;
-        if (connectionPool.isEmpty() && totalConnections() < maxConnectionsNumber) {
-            connection = createConnection();
-            logger.log(Level.TRACE, "New connection created");
-        } else {
-            try {
-                connection = connectionPool.poll(requestTimeoutValue, requestTimeoutUnit);
-                logger.log(Level.TRACE, "Connection obtained from connection pool");
-            } catch (InterruptedException e) {
-                logger.log(Level.WARN, "Cannot get connection from Connection pool, new connection will be " +
-                        "created", e);
+    public synchronized Connection getConnection() throws SQLException, IllegalStateException {
+        if (connectionPool == null || activeConnections == null) {
+            throw new IllegalStateException("Connection pool is shut down");
+        }
+        try {
+            PoolConnection connection = connectionPool.poll(requestTimeoutValue, requestTimeoutUnit);
+            logger.log(Level.TRACE, "Connection obtained from connection pool");
+            if (connection == null) {
```

"poll method returns the head of this queue, or null if the specified waiting time elapses before an element is available" - need to raise an exception is can't get a connection from pool per time out - opening new can lead to the problem.